### PR TITLE
EE-1186: persist bids directly in state trie

### PR DESCRIPTION
**Pull request by henrytill**
This PR removes the `Bids` map stored under the auction's `BIDS_KEY` named key and instead persists bids directly in the state trie under `Key::Bid` keys.  It also adds `RpcRequest::GetBids` (and related) in service of `get-auction-info`.

### CODE REVIEW

**Review by EdHastingsCasperLabs — APPROVED** [review]

File: types/src/system/auction/detail.rs [hunk 1]
```diff
@@ -270,21 +292,27 @@ pub fn reinvest_delegator_rewards(
...
-pub fn reinvest_validator_reward(
-    bids: &mut Bids,
+pub fn reinvest_validator_reward<P>(
+    provider: &mut P,
     seigniorage_allocations: &mut Vec<SeigniorageAllocation>,
     validator_public_key: PublicKey,
     amount: U512,
-) -> Result<Option<URef>, Error> {
-    let bid = match bids.get_mut(&validator_public_key) {
+) -> Result<Option<URef>, Error>
```
Comment (mpapierski) [inline types/src/system/auction/detail.rs:306]: No need for Option anymore

**Review by mpapierski — APPROVED** [review]

File: types/src/execution_result.rs [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 use alloc::{
     format,
+    prelude::v1::Box,
```
[the remark this answers is not in the document]
Comment (henrytill) [reply types/src/execution_result.rs]: Fixed

File: types/src/system/auction/detail.rs (hunk 1 above)
Comment (henrytill) [reply types/src/system/auction/detail.rs]: Fixed

**Pull request merged**